### ISSUE TITLE
ios.py get_facts bgp neighbours producing incorrect prefix counts

### ISSUE THREAD

**Issue by lukerussell** [napalm-automation/napalm#226]
Hi,

get_facts for bgp_neighbors is producing incorrect results for the prefix counts with my version of IOS/XE. Images are provided by VIRL.

I strongly suspect this is related to the regex match patterns, but I don't know enough to offer a solution.

Sample output from IOSv 15.6(1)T:

`    Prefixes Current:               4         12 (Consumes 960 bytes)`
 ```
 Local Policy Denied Prefixes:    --------    -------
    AS_PATH loop:                       n/a          4
    Bestpath from this peer:             12        n/a
    Total:                               12          4
```

get_facts produces this:
```
ok: [2A-R1] => {
    "result": {
        "ansible_facts": {
            "bgp_neighbors": {
                "global": {
                    "peers": {
                        "10.130.128.2": {
                            "address_family": {
                                "ipv4": {
                                    "accepted_prefixes": 4, 
                                    "received_prefixes": 8, 
                                    "sent_prefixes": 4
                                }
                            }, 
                            "description": "", 
                            "is_enabled": true, 
                            "is_up": true, 
                            "local_as": 65200, 
                            "remote_as": 65530, 
                            "remote_id": "192.168.0.13", 
                            "uptime": 3807
                        }, 

```

Sample output from IOS-XEv 03.17.00.S (15.6(1)S):

`    Prefixes Current:               2         14 (Consumes 1680 bytes)`
````
  Local Policy Denied Prefixes:    --------    -------
    AS_PATH loop:                       n/a          2
    Bestpath from this peer:             14        n/a
    Total:                               14          2

```

get_facts produces this:
```
ok: [1A-R1] => {
    "result": {
        "ansible_facts": {
            "bgp_neighbors": {
                "global": {
                    "peers": {
                        "10.130.0.14": {
                            "address_family": {
                                "ipv4": {
                                    "accepted_prefixes": 2, 
                                    "received_prefixes": 4, 
                                    "sent_prefixes": 2
                                }
                            }, 
                            "description": "", 
                            "is_enabled": true, 
                            "is_up": true, 
                            "local_as": 65100, 
                            "remote_as": 65530, 
                            "remote_id": "192.168.0.13", 
                            "uptime": 5381
                        }
                    }, 
```

**Comment by dbarrosop** [comment]
Could you share the output of the following commands?

- `show ip bgp summary`
- `show ip bgp summary | begin Neighbor `
- `show ip bgp neighbors 10.130.0.14 | inc router ID ` (replace the IP if necessary)
- `show ip bgp neighbors 10.130.0.14 | inc Prefixes Current`  (replace the IP if necessary)
- `show ip bgp neighbors 10.130.0.14 | section Local Policy`   (replace the IP if necessary)

Thanks!

**Comment by lukerussell** [comment]
Hi @dbarrosop,

Requested outputs below from the IOS-XE router.

```
1A-R1#show ip bgp summary
BGP router identifier 192.168.0.17, local AS number 65100
BGP table version is 17, main routing table version 17
16 network entries using 3968 bytes of memory
16 path entries using 1920 bytes of memory
4/4 BGP path/bestpath attribute entries using 1024 bytes of memory
3 BGP AS-PATH entries using 104 bytes of memory
0 BGP route-map cache entries using 0 bytes of memory
0 BGP filter-list cache entries using 0 bytes of memory
BGP using 7016 total bytes of memory
BGP activity 16/0 prefixes, 17/1 paths, scan interval 60 secs

Neighbor        V           AS MsgRcvd MsgSent   TblVer  InQ OutQ Up/Down  State/PfxRcd
10.130.0.14     4        65530      24      22       17    0    0 00:15:48       14
1A-R1#show ip bgp summary | begin Neighbor
Neighbor        V           AS MsgRcvd MsgSent   TblVer  InQ OutQ Up/Down  State/PfxRcd
10.130.0.14     4        65530      24      22       17    0    0 00:15:54       14
1A-R1#show ip bgp neighbors 10.130.0.14 | inc router ID
  BGP version 4, remote router ID 192.168.0.13
1A-R1#show ip bgp neighbors 10.130.0.14 | inc Prefixes Current
    Prefixes Current:               2         14 (Consumes 1680 bytes)
1A-R1#show ip bgp neighbors 10.130.0.14 | section Local Policy
  Local Policy Denied Prefixes:    --------    -------
    AS_PATH loop:                       n/a          2
    Bestpath from this peer:             14        n/a
    Total:                               14          2
```

```
TASK [get_facts : Print facts data] ********************************************
ok: [1A-R1] => {
    "result": {
        "ansible_facts": {
            "bgp_neighbors": {
                "global": {
                    "peers": {
                        "10.130.0.14": {
                            "address_family": {
                                "ipv4": {
                                    "accepted_prefixes": 2, 
                                    "received_prefixes": 4, 
                                    "sent_prefixes": 2
                                }
                            }, 
                            "description": "", 
                            "is_enabled": true, 
                            "is_up": true, 
                            "local_as": 65100, 
                            "remote_as": 65530, 
                            "remote_id": "192.168.0.13", 
                            "uptime": 838
                        }
                    }, 
                    "router_id": "192.168.0.17"
                }
            }
        }, 
        "changed": false
    }
}
```

**Comment by ktbyers** [comment]
@lukerussell I have submitted a fix for this. See https://github.com/napalm-automation/napalm/pull/227/files.

Can you test that this fix resolves your issue?

**Comment by ktbyers** [comment]
Fixed here https://github.com/napalm-automation/napalm-ios/pull/3

**Comment by dbarrosop** [comment]
I released napalm-ios 0.1.2 with the bugfixes, Unfortunately people can't take advantage of the new code until we merge #232 but I guess that will be happening next week the latest (unless we discover something horrible in my PR : P)
